src/appengine/handlers/external_update.py

```diff
@@ -70,31 +86,57 @@ def handle_update(testcase, revision, stacktrace, error):
   # not run).
   data_handler.record_fuzz_target(fuzz_target.engine, fuzz_target.binary,
                                   testcase.job_type)
-
-  state = stack_analyzer.get_crash_data(
-      stacktrace,
-      fuzz_target=fuzz_target_name,
-      symbolize_flag=False,
-      already_symbolized=True,
-      detect_ooms_and_hangs=True)
-  crash_comparer = CrashComparer(state.crash_state, testcase.crash_state)
-  if not crash_comparer.is_similar():
+  states = [
```

Comment:
There's a bit of complexity in the way you are handling these multiple values. One cleaner way to do this is to just factor out the existing code for validating a single stacktrace into its own function, e.g. `is_still_crashing`, and just call this in a loop in this function. We can also early exit on the first true result, and avoid having to do crash comparisons and other analysis all up front.

i.e.

```python
...
data_handler.record_fuzz_target(fuzz_target.engine, fuzz_target.binary,
                                  testcase.job_type)
for stacktrace in stacktraces:
  if is_still_crashing(testcase, stacktrace):
    logs.log(f'{testcase.key.id()} still crashes.')
    return

# If we get here, it means that every single trial resulted in a non-crash. Close the testcase.
logs.log(f'No matching crash detected in {len(stacktraces)} trials, marking as fixed.')
_mark_as_fixed(testcase, revision)
...
```